src/Microsoft.AspNetCore.Authentication.Facebook/FacebookHandler.cs

```diff
@@ -35,7 +36,10 @@ public FacebookHandler(HttpClient httpClient)
             }
 
             var response = await Backchannel.GetAsync(endpoint, Context.RequestAborted);
-            response.EnsureSuccessStatusCode();
+            if (!response.IsSuccessStatusCode)
+            {
+                throw new HttpRequestException($"Failed to retrived Facebook user information ({response.StatusCode}) Please check if the authentication information is correct and the corresponding Facebook API is enabled.");
```

Comment:
Throw exception here is not ideal, but without breaking change, there is no other means to return the information to outer logic. Returns `null` can indicate an error, but the message composed here is more readable.

I think there is room for us to improve the error handling but it could be out of the scope of this PR. 

I'll update the comments, this is actually `Facebook Graph API`.